datasheet/Zubax_GNSS2_Datasheet.tex

```diff
@@ -0,0 +1,777 @@
+
+If the YMODEM protocol is used, the file name field in the transfer header packet will be ignored.
+
+There are heaps of software products and scripts that support these file transfer protocols. For instance, the popular program \textbf{sz} (available on most Linux distributions) can be used as follows:\\
+\texttt{sz -vv --ymodem --1k \$file > \$port < \$port}
+
+\section{Via CAN bus}
+
+The bootloader supports the UAVCAN firmware update protocol. Please refer to the UAVCAN specification for theory.
+
+The bootloader only utilizes the primary CAN interface, which is CAN1. The redundant interface CAN2 remains in the passive mode while the bootloader is running. It is therefore required that if there is only one interface in use, it must be CAN1.
+
+The following table describes the mapping from the bootloader states to the UAVCAN node status codes:
+
+\begin{ZubaxSimpleTable}{Bootloader states and UAVCAN node status codes}{| X | X | X |}
+Bootloader state & UAVCAN node mode & UAVCAN node health\\
+NoAppToBoot & SOFTWARE{\_}UPDATE & ERROR\\
+BootDelay, ReadyToBoot & MAINTENANCE & OK\\
+BootCancelled & MAINTENANCE & WARNING\\
+AppUpgradeInProgress & SOFTWARE{\_}UPDATE & OK
+\end{ZubaxSimpleTable}
+
+The vendor specific status code of the node status message contains the status code of the last attempt to upgrade the firmware. Please refer to the error code table provided above.
+
+\clearpage
+\subsection{Supported messages}
+\begin{ZubaxSimpleTable}{Supported messages}{| l | l | l | l |}
+Data type & Direction & Period & Transfer priority \\
+\texttt{uavcan.protocol.NodeStatus} & Output & 1 Hz & 24(low) \\
+\texttt{uavcan.protocol.debug.LogMessage} & Output & Aperiodic & 31(lowest) \\
+\texttt{uavcan.protocol.dynamic{\_}node{\_}id.Allocation} & Input/Output & Aperiodic & 24 (low)
+\end{ZubaxSimpleTable}
+
+\begin{ZubaxSimpleTable}{Supported messages annotation}{| l | X |}
+Data type & Note\\
+\texttt{uavcan.protocol.NodeStatus} & Refer to the bootloader state mapping table.\\
+\texttt{uavcan.protocol.debug.LogMessage} & Used to report the application upgrade progress and status in a human readable form.\\
+\texttt{uavcan.protocol.dynamic{\_}node{\_}id.Allocation} & Used only during initialization, if the application did not provide a specific node ID to use.
+\end{ZubaxSimpleTable}
+\clearpage
+\subsection{Supported services}
+
+Servers: 
+
+\begin{ZubaxSimpleTable}{Supported servers}{| l | X |}
+Data type & Note\\
+uavcan.protocol.GetNodeInfo & The nested structure  \textbf{uavcan.protocol.SoftwareVersion} is populated only if there is a valid application.\\
+uavcan.protocol.RestartNode & Restarts the bootloader.\\
+uavcan.protocol.file.BeginFirmwareUpdate & Initiates the firmware update process, unless it was initiated earlier before the application has rebooted into the bootloader.
+\end{ZubaxSimpleTable}
+
+Clients:
```

Comment:
What is the purpose of this statement if the table is already captioned? Let us not be redundant.